src/test/java/com/skalashynski/spring/springboot/service/StudentServiceTest.java

```diff
@@ -0,0 +1,231 @@
+package com.skalashynski.spring.springboot.service;
+
+import com.skalashynski.spring.springboot.entity.Student;
+import com.skalashynski.spring.springboot.exception.ApiException;
+import com.skalashynski.spring.springboot.repository.StudentRepository;
+import com.skalashynski.spring.springboot.service.impl.StudentServiceImpl;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+import static org.mockito.BDDMockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class StudentServiceTest {
+
+    @Mock
+    private StudentRepository studentRepository;
+
+    @InjectMocks
+    private StudentServiceImpl studentService;
+
+    private static Stream<Student> provideStudents() {
+        return Stream.of(
+            new Student(Long.MIN_VALUE, "firstName", "lastName",
+                        LocalDate.of(1993, 1, 17), LocalDateTime.now()),
+            new Student(Long.MAX_VALUE, "someName", "lastName",
+                        LocalDate.of(2000, 12, 1), LocalDateTime.now()),
+            new Student(1L, "", "",
+                        LocalDate.of(1980, 5, 28), LocalDateTime.now()),
+            new Student(20L, "Ivan", null,
+                        LocalDate.of(2005, 8, 10), LocalDateTime.now())
+        );
+    }
+
+    private static Stream<Arguments> provideArgsForUpdateTest() {
+        return Stream.of(
+                arguments(1L, new Student(20L, "Ivan", null,
+                        LocalDate.of(2000, 1, 1), LocalDateTime.now())),
+                arguments(Long.MIN_VALUE, new Student(Long.MAX_VALUE, "someName", "lastName",
+                        LocalDate.of(2000, 12, 1), LocalDateTime.now()))
+        );
+    }
```

Comment:
Optional. For best practices ,you can move all test parameters to separate new `Const` class.

Reply:
I did by creating separate `class StudentServiceTestArgsProvider` and I have 2 questions:

1. Should I put methods in this class to provide primitive arguments (for example, for the `getByIdTest(long id)` method) or was it better to leave `@ValueSource(longs ...)` for such methods?
2. Is it normal that I have a huge path in @MethodSource that is repeated for all methods? I thought how to fix it, but it didn't work.

Reply:
1. It's only preferences. Depends on the project volume and so on. No restrictions, no preferences. Up to you.
2. It's cons of Junit. Have a look at Spock.